config/routes.rb

```diff
@@ -1,3 +1,8 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  get 'main_page/home'
+  get 'products/new'
+  get 'products/show'
+  get 'products/edit'
```

Comment:
if you define `resources :products` you don't need to explicitly define these get routes for products since `resources` will create those for you